scripts/download_artifacts.py

```diff
@@ -272,8 +285,98 @@ def issue_hashes(repo_name: str, token: str):
     hashes = [issue['title'].split(' ')[-1] for issue in issues if 'pull_request' not in issue.keys()]
     return hashes
 
+def search_and_download_previous_report_artifact(
+    artifact_name_template:str, previous_hash: str, prev_commits: List[str], repo_name: str, token: str
+):
+    """Download a most recent previous report artifact and return the corresponding hash.
+       Return None if no corresponding hash has been found
+    """
+    artifact_name_template += "-report.log"
+
+    # check if we already have a previous artifact available
+    # mostly for regenerate issues
+    if previous_hash:
+        name_components = artifact_name_template.split("-")
+        name_components[4] = "{}"
+        previous_name = "-".join(name_components).format(previous_hash)
+        name_components[4] = "[^-]*"
+        name_regex = "-".join(name_components)
+        dir_contents = " ".join(os.listdir("./previous_logs"))
+        possible_previous_logs = re.findall(name_regex, dir_contents)
+        if len(possible_previous_logs) > 1:
+            print(
+                f"found more than 1 previous log for {name_regex}: {possible_previous_logs}"
+            )
+            for log in possible_previous_logs:  # remove non-recent logs
+                if previous_name not in log:
+                    print(f"removing {log} from previous_logs")
+                    os.remove(os.path.join("./previous_logs", log))
+            return previous_hash
+        if len(possible_previous_logs) == 1:
+            print(
+                f"found single log: {possible_previous_logs[0]}. Skipping download"
+            )
+            return previous_hash    
+
+    # download previous artifact
+    base_hash, base_id = get_valid_artifact_hash(
+        prev_commits, repo_name, token, artifact_name_template
+    )
+    if base_hash != "No valid hash":
+        artifact_zip = download_artifact(
+            artifact_name_template.format(base_hash), str(base_id), token, repo_name, "./temp/"
+        )
+        extract_artifact(
+            artifact_zip,
+            outdir="previous_logs",
+        )
+        return base_hash
+
+    print(
+        f"found no valid hash for {artifact_name_template}. possible hashes: {prev_commits}"
+    )
+    return None
+
+def download_build_log_artifact(
+    artifact_name_template:str, base_hash: str, repo_name: str, token: str, output_dir: str
+):
+    """Download the build log zipfiles for a corresponding base_hash and extract them to output_dir"""
+    # input validation
+    if not base_hash:
+        return
```

Comment:
Can we throw an error instead?

Reply:
Since report log download doesn't throw an error if the base_hash is not found, I made it not to throw an error to match the behavior. Should I modify both places then?

Reply:
Ah, I misread this as user-input validation. A log statement should be good here.
```suggestion
        print("Missing base hash, skipping downloading the build log artifact")
        return
```